explorer/interpreter/pattern_match.h

```diff
@@ -26,12 +26,11 @@ class Arena;
 // PatternMatch call and then merged into an existing scope on success.
 // The matches for generic variables in the pattern are output in
 // `generic_args`.
-[[nodiscard]] auto PatternMatch(Nonnull<const Value*> p, ExpressionResult v,
```

Comment:
This one seems worth keeping: the return value is easily overlooked because the function isn't named as though it returns a value, and disregarding the return value is almost certainly a bug.

Reply:
Reverted on the basis that explorer is archived, so I'm not going to feel bad having it have readability issues.